File: src/main/java/com/github/bpark/companion/FusekiUpload.java
package com.github.bpark.companion;

import org.eclipse.rdf4j.repository.Repository;
import org.eclipse.rdf4j.repository.RepositoryConnection;
import org.eclipse.rdf4j.repository.sparql.SPARQLRepository;
import org.eclipse.rdf4j.rio.RDFFormat;

import java.io.File;
import java.io.IOException;

public class FusekiUpload {

    public static void main(String[] args) {
        String sparqlEndpoint = "http://192.168.56.8:8080/demo/update";
        Repository repo = new SPARQLRepository(sparqlEndpoint);
        repo.initialize();


        try (RepositoryConnection conn = repo.getConnection()) {
            conn.add(new File("data/skos-demo.ttl"), null, RDFFormat.TURTLE);
        } catch (IOException e) {
            throw new RuntimeException(e);
        }

    }
}

File: src/main/kotlin/com/github/bpark/companion/FusekiUpload.kt
package com.github.bpark.companion

import org.eclipse.rdf4j.repository.sparql.SPARQLRepository
import org.eclipse.rdf4j.rio.RDFFormat
import java.io.File
import java.io.IOException

fun main(args: Array<String>) {
    val sparqlEndpoint = "http://192.168.56.8:3030/demo/update"
    val repo = SPARQLRepository(sparqlEndpoint)
    repo.initialize()


    try {
        repo.connection.use { conn -> conn.add(File("data/foaf-demo.ttl"), null, RDFFormat.TURTLE) }
    } catch (e: IOException) {
        throw RuntimeException(e)
    }

}